### DISCUSSION TITLE
Share Collection Content Table with Public

### DISCUSSION THREAD

**Discussion by swallace21** [directus/directus#20984] [Feature Requests]
### Summary

Allow admins to grant role-based CRUD permissions for the Public to access the Content Table view. Google Sheets allows anonymous users read-only or full CRUD permissions for spreadsheets. Allowing admins to share specific Content Tables would help Directus further achieve its goal of democratizing data. Crowdsourcing platforms, such as Amazon Mechanical Turk and Prolific, explicitly warn against forcing crowdworkers to sign up for accounts on external platforms to contribute.[](url)

### Motivation

This extension of the existing Sharing feature will allow admins to publicly share data and motivate contributions to edit and maintain the data from the public. Google Sheets has been used in the past to crowdsource datasets rapidly,  for example:

The little spreadsheet that could and did: crowdsourcing COVID-19, higher education, data, and stories
https://bryanalexander.org/research-topics/the-little-spreadsheet-that-could-and-did-crowdsourcing-covid-19-higher-education-data-and-stories/

### Detailed Design

The Content Table view's right sidebar would contain the Shares tab.
This will allow admins to use the existing Shares UI options to share the full table from the Content view per Content.
The models reused can be derived from sharing API access to viewing the Table UI as well.

### Must Have:
- Allow the Sharing feature to work with the Content Table View.
- Allow the public to view the Content Table View and then add, update, or delete individual rows.

### Should Have:
- Provide each anonymous user with a different unique ID. So, their edit history is correctly represented.

### Could Have:
- The ability to only specific columns in the table to be edited.
- Allow admins to enable/disable Import/Export CSV.

### Won't Have:
- In-line editing of table cells.

### Drawbacks

The complexity should not be high, since much of the workflow and UI can be reused from the existing Sharing UI.
This should be able to be implemented in user space with minimal migrations on the backend.
The documentation for this feature can go into the existing Sharing documentation.
This will not be a breaking change because Sharing across various data types is off by default.

### Alternatives

Using Google Sheets. <- Very poor control and lacks certain automation features. Is not backed by an actual database.
I have also built my own solutions in the past. Integrating this with an existing platform would help others more, and decrease the maintenance burden.<- Very poor control and lack of certain automation features.<- Very poor control and lack of certain automation features.

### Adoption Strategy

Implementing Sharing and Permissions to edit an entire Collection's Table would be opt-in. The UI implementation could be very similar to the existing Sharing UI. Thus, learnability should be easy and existing users would have to opt-in.Thus, learnability should be easy, and existing users would have to opt in.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
